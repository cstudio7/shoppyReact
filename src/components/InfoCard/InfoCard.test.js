import React from 'react';
import { shallow } from 'enzyme';
import AdminCards from './index';

describe('Test the AdminCards Component', () => {
  beforeEach(() => {
    localStorage.setItem(
      'Authentication',
      'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXlMb2FkIjp7ImlkIjoiN2NhM2JjYzktZTMzOS00OTcwLTk0YjMtYjI5YTg1N2ViM2ExIiwicm9sZSI6W119LCJpYXQiOjE1NTE0NjE4OTcsImV4cCI6MTU1MTU0ODI5N30.IzGf0L_pH9W_gN2AC2Ee5aWqwSO4l3fa46ZrcyUbMKU'
    );
  });
  afterEach(() => {
    localStorage.removeItem('Authentication');
  });

  it('should match snapshot', () => {
    const userRole = 'admin';
    const product = 'product';
    const wrapper = shallow(<AdminCards userRole={userRole} currentPage={product} />);
    wrapper.find('p').simulate('click');
    expect(wrapper.state('open')).toBe(true);
    wrapper.instance().onCloseModal();
    wrapper.instance().onOpenModal();
    wrapper.instance().displayAdminButtons();
    wrapper.instance().displayAddToCartButton();
  });
  it('should match snapshot', () => {
    const userRole = 'attendant';
    const product = 'product';
    const wrapper = shallow(<AdminCards userRole={userRole} currentPage={product} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Test the AdminCards Component', () => {
  it('should redirect to login', () => {
    const userRole = 'attendant';
    const product = 'product';
    const wrapper = shallow(<AdminCards userRole={userRole} currentPage={product} />);
    expect(wrapper).toMatchSnapshot();
  });
});
