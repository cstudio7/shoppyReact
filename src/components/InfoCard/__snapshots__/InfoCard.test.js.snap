// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the AdminCards Component should match snapshot 1`] = `
<Fragment>
  <Modal
    animationDuration={500}
    blockScroll={true}
    center={true}
    classNames={Object {}}
    classes={
      Object {
        "closeButton": "styles_closeButton__20ID4",
        "closeIcon": "styles_closeIcon__1QwbI",
        "modal": "styles_modal__gNwvD",
        "overlay": "styles_overlay__CLSq-",
        "overlayCenter": "styles_overlayCenter__YHoO7",
        "transitionEnter": "styles_transitionEnter__3j_-a",
        "transitionEnterActive": "styles_transitionEnterActive___eQs7",
        "transitionExit": "styles_transitionExit__1KmEf",
        "transitionExitActive": "styles_transitionExitActive__1nQXw",
      }
    }
    closeIconId={null}
    closeIconSize={28}
    closeIconSvgPath={
      <path
        d="M28.5 9.62L26.38 7.5 18 15.88 9.62 7.5 7.5 9.62 15.88 18 7.5 26.38l2.12 2.12L18 20.12l8.38 8.38 2.12-2.12L20.12 18z"
      />
    }
    closeOnEsc={true}
    closeOnOverlayClick={true}
    focusTrapOptions={Object {}}
    focusTrapped={false}
    modalId={null}
    onClose={[Function]}
    onEntered={null}
    onEscKeyDown={null}
    onExited={null}
    onOverlayClick={null}
    open={false}
    overlayId={null}
    showCloseIcon={true}
    styles={Object {}}
  >
    <ProductForm />
  </Modal>
  <div
    className="categoryCardContainer"
  >
    <Card
      className="categoryCard"
    >
      <Image
        as="img"
        src=""
        ui={true}
      />
      <CardContent
        extra={true}
      >
        <CardHeader
          className="adminCardTitle"
        />
        <CardDescription
          className="adminCardDescription"
        />
        <CardDescription
          className="adminCardDescription"
        />
        <Button
          animated={true}
          as="button"
        >
          <ButtonContent
            visible={true}
          >
            Add to Cart
          </ButtonContent>
          <ButtonContent
            hidden={true}
          >
            <Icon
              as="i"
              name="arrow right"
            />
          </ButtonContent>
        </Button>
      </CardContent>
    </Card>
  </div>
</Fragment>
`;

exports[`Test the AdminCards Component should redirect to login 1`] = `
<Redirect
  push={false}
  to="/login"
/>
`;
